pennylane/devices/task_qubit.py

```diff
@@ -0,0 +1,246 @@
+    ...     def circuit(x):
+    ...         qml.RX(x[0], wires=0)
+    ...         qml.RY(x[1], wires=0)
+    ...         qml.RZ(x[2], wires=0)
+    ...         qml.RZ(x[0], wires=1)
+    ...         qml.RX(x[1], wires=1)
+    ...         qml.RY(x[2], wires=1)
+    ...         return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+    ...     weights = tf.Variable(qml.numpy.random.rand(3))
+    ...     def f_submit(weights):
+    ...         with tf.GradientTape() as tape:
+    ...             # Use the circuit to calculate the loss value
+    ...             loss = tf.abs(circuit(weights)-0.5)**2
+    ...         return tape.gradient(loss, weights)
+
+    ...     print(client.submit(f_submit, weights).result())
+    tf.Tensor([0.01776833 0.05199685 0.03689981], shape=(3,), dtype=float64)
+
+    Args:
+        wires (int): The number of wires to initialize the device with.
+        backend (None, str): Indicates the PennyLane device type to use for offloading
+            computation tasks. The TensorFlow and PyTorch interfaces are the preferred types
+            for gradient computations. This is due to existing support in Dask for
+            TF and Torch datatypes.
+        future (None, bool): Indicates whether the internal circuit evaluation returns a future
+            to a result. This allows building of dependent workflows, but currently only works with
+            explicit calls to `device.batch_execute` with a PennyLane native device type such as
+            (`default.qubit`, `lightning.qubit`).
+        gen_report: Indicates whether the backend task-scheduler will generate a performance report based on the tasks that were run.
+    """
+
+    operations = DefaultQubit.operations
+    observables = DefaultQubit.observables
+
+    name = "Task-based proxy PennyLane plugin"
+    short_name = "task.qubit"
+    pennylane_requires = __version__
+    version = __version__
+    author = "Xanadu Inc."
+
+    supported_devices = {
+        "default.qubit",
+        "default.qubit.tf",
+        "default.qubit.jax",
+        "default.qubit.torch",
+        "lightning.qubit",
+    }
+
+    def __init__(
+        self,
+        wires,
+        *,
+        shots=None,
+        analytic=None,
+        backend="default.qubit",
+        gen_report: Union[bool, str] = False,
+        future=False,
+    ):
+        super().__init__(0, shots, cache=False)
+
+        self._backend = backend
+        self._backend_cls = qml.plugin_devices[backend].load()
+        self._backend_dev = self._backend_cls(wires=0)
+        self._wires = wires
+        self._gen_report = gen_report
+        self.num_wires = wires if isinstance(wires, int) else len(wires)
+        self._future = future
+        self.expand_fn = self.default_expand_fn
+
+        if backend not in TaskQubit.supported_devices:
+            raise Exception("Unsupported device backend.")
+
+    def __get__(self, obj, objtype=None):
```

Comment:
What does this method do?

Reply:
This was needed as part of allowing the device to return the supported functionality from the provided backend. Since this class aims to be just a proxy interface, the __get__ method follows the [Python descriptor](https://docs.python.org/3/howto/descriptor.html) guidelines, and should offload to the backend devices method when instantiated.